application/models/services/SurveyUpdater/GeneralSettings.php

```diff
@@ -313,7 +313,7 @@ private function setField($field, $input, Survey $survey, $meta, $fieldOpts = nu
             case static::FIELD_TYPE_DATETIME:
                 $value = !empty($value)
                     ? $this->formatDateTimeInput($value)
-                    : $default;
+                    : null;
```

Comment:
We are applying a general change here while the bug is for couple of fields.
Should we do so?

Have you tried reviewing `$fieldOpts['default']` ?
Maybe we just need to change the options passed to the method...?

Maybe the method doesn't allow a mechanism for "clearing" the values?

Reply:
> We are applying a general change here while the bug is for couple of fields. Should we do so?
> 
> Have you tried reviewing `$fieldOpts['default']` ? Maybe we just need to change the options passed to the method...?
> 
> Maybe the method doesn't allow a mechanism for "clearing" the values?

The issue is here: https://github.com/LimeSurvey/LimeSurvey/blob/3eac536c0bc6f4ac55458d4be6b6af1ced787123/application/models/services/SurveyUpdater/GeneralSettings.php#L300 the check for empty against null means that the value is set to "initvalue". initvalue is the current value of the field so it is never overwritten by null.

Reply:
Also it appears nothing else uses the field type datetime and it makes sense to write null when null is given.